src/api/app/models/worker_status.rb

```diff
@@ -7,6 +7,13 @@ def hidden
       hidden_projects(ws)
     end
 
+    def statistics
+      mydata = Rails.cache.read('workerstatus')
+      ws = Nokogiri::XML(mydata || Backend::Api::BuildResults::Worker.status)
+      { building: ws.search('//building').size, waiting: ws.search('//waiting').size,
+        blocked: ws.search('//blocked').size, idle: ws.search('//idle').size }
```

Comment:
For `building` and `idle` is fine to count the ocurrences in the xml with `size`. For `waiting` and `blocked`, shouldn't we sum the `jobs` attributes?

Reply:
Don't calculate it when recording. I would send it as `waiting` with the `arch` as tag and the `jobs` as field.

Reply:
summarizing you can do in the visualization

Reply:
hm, i read it too late @hennevogel :rofl: 

If you want like that I would separate it in another PR and in this PR just keep the workers (idle x building) and then do another PR to handle jobs.

Reply:
@hennevogel could you please take a look again? I added the waiting and jobs fields.